packages/compartment-mapper/src/parse-pre-cjs.js

```diff
@@ -28,37 +28,59 @@ export const parsePreCjs = async (
   );
 
   /**
-   * @param {Object} moduleExports
+   * @param {Object} moduleEnvironmentRecord
    * @param {Compartment} compartment
    * @param {Record<string, string>} resolvedImports
    */
-  const execute = async (moduleExports, compartment, resolvedImports) => {
+  const execute = (moduleEnvironmentRecord, compartment, resolvedImports) => {
     const functor = compartment.evaluate(source);
 
+    const originalExports = new Proxy(
+      Object.create(compartment.globalThis.Object.prototype),
+      {
+        get(target, prop) {
+          return moduleEnvironmentRecord[prop] || target[prop]; // this makes things like exports.hasOwnProperty() work.
+        },
+        set(target, prop, value) {
+          moduleEnvironmentRecord[prop] = value;
+          return true;
+        },
+      },
+    );
+
+    let finalExports = originalExports;
+
     const module = freeze({
       get exports() {
-        return moduleExports;
+        return finalExports;
       },
       set exports(value) {
-        moduleExports.default = value;
+        finalExports = value;
       },
     });
 
     const require = freeze((/** @type {string} */ importSpecifier) => {
       const namespace = compartment.importNow(resolvedImports[importSpecifier]);
       if (namespace.default !== undefined) {
-        return namespace.default;
+        if (Object.keys(namespace).length > 1) {
+          return { ...namespace.default, ...namespace }; // this resembles Node's behavior more closely
```

Comment:
This comment could use more elaboration.

Reply:
Oh I love elaborating. Had to restrain myself to leave it at this.